pkg/bootstrap/platform/gcp_test.go

```diff
@@ -183,13 +209,40 @@ func TestGCPMetadata(t *testing.T) {
 			for e, v := range tt.env {
 				os.Setenv(e, v)
 			}
-			once = sync.Once{}
-			mg := gcpEnv{tt.shouldFill, tt.projectIDFn, tt.numericProjectIDFn, tt.locationFn, tt.clusterNameFn, tt.instanceIDFn,
-				tt.instanceTemplateFn, tt.instanceCreatedByFn}
-			got := mg.Metadata()
+			envOnce = sync.Once{}
+			e := gcpEnv{}
+			shouldFillMetadata, projectIDFn, numericProjectIDFn, clusterLocationFn, clusterNameFn, instanceNameFn, instanceIDFn, instanceTemplateFn, createdByFn =
+				tt.shouldFill, tt.projectIDFn, tt.numericProjectIDFn, tt.locationFn, tt.clusterNameFn, tt.instanceNameFn, tt.instanceIDFn, tt.instanceTemplateFn,
+				tt.instanceCreatedByFn
+			got := e.Metadata()
 			if !reflect.DeepEqual(got, tt.want) {
 				t.Errorf("gcpEnv.Metadata() => '%v'; want '%v'", got, tt.want)
 			}
 		})
 	}
 }
+
+func TestCacheMetadata(t *testing.T) {
```

Comment:
This is a good test of this function. In general, however, we should strive to avoid adding tests for unexported methods. It is better to test the "public" APIs and not worry about the the implementation details.

Caching is a bit of an internal detail. However, I can understand the desire to validate the behavior. Maybe we could call `Metadata()` and then `Labels()` and check counters in supplied funcs to verify the metadataFns were only called once?

I think we might want to consider having `Labels` take a labelLookupFn or whatever to allow for easier validation of that behavior. That would allow us to test other aspects of `Labels`, including timeouts, etc., too.

Thoughts?